docs/custom_promise_types/modules.md

```diff
@@ -0,0 +1,363 @@
+# CFEngine custom promise types
+
+Custom promise types can be added as _Promise modules_.
+These enable users to implement their own promise types for managing new resources.
+They can be added without a binary upgrade, and be shared between users.
+
+## Using custom promise types
+
+A new top level `promise` block (similar to `body`) is used to add new promise types.
+After adding a promise type, it is used in the same way as built-in promise types, there is no special syntax.
+
+### Example of using a promise module
+
+```cf3
+promise agent git
+{
+  interpreter => "/usr/bin/python3";
```

Comment:
Curious, what would you set this to if you had some C binary implementing the module?

Reply:
@nickanderson I think it makes the most sense to skip it then, don't specify an interpreter if you don't want interpretation (and it would default to empty string).